runtime/src/launch/smp/launch-smp.c

```diff
@@ -19,20 +19,17 @@
 
 #include <stdio.h>
 #include <string.h>
+#include "chpl-env.h"
 #include "chpllaunch.h"
 #include "error.h"
 
 
 // Simple launcher that just sets GASNET_PSHM_NODES and launchers the _real
 
 int chpl_launch(int argc, char* argv[], int32_t numLocales) {
-  char numlocalesval[12]; // big enough for int32_t
   char baseCommand[4096];
 
-  snprintf(numlocalesval, sizeof(numlocalesval), "%d", (int)numLocales);
-  if (setenv("GASNET_PSHM_NODES", numlocalesval, 1) != 0) {
-    chpl_error("Cannot setenv(\"GASNET_PSHM_NODES\")", 0, 0);
-  }
+  chpl_env_set_uint("GASNET_PSHM_NODES", numLocales, 1);
```

Comment:
This is a discussion for another time and certainly not something for this PR, but I wonder if we should automatically enable oversubscription or smp or udp if we detect local spawning.

Reply:
Yeah, if we can dependably detect that we're going to oversubscribe then that would be a nice thing, user experience-wise.

Reply:
As a stepping stone, maybe check for GASNET_SPAWNFN in the docs dir (to look for GASNET_SPAWNFN=L) and look at smp documentation for places to add a cross-reference?